crates/storage-query-datafusion/src/journal/table.rs

```diff
@@ -65,25 +63,48 @@ pub(crate) enum ScannedEntry {
 #[derive(Debug, Clone)]
 struct JournalScanner;
 
-impl ScanLocalPartition for JournalScanner {
+impl ScanLocalPartitionInPlace for JournalScanner {
     type Builder = SysJournalBuilder;
-    type Item = (JournalEntryId, ScannedEntry);
+    type Item<'a> = (JournalEntryId, ScannedEntry);
 
-    fn scan_partition_store(
+    fn for_each_row<F: for<'a> FnMut(Self::Item<'a>) -> ControlFlow<()> + Send + Sync + 'static>(
         partition_store: &PartitionStore,
         range: RangeInclusive<PartitionKey>,
-    ) -> Result<impl Stream<Item = restate_storage_api::Result<Self::Item>> + Send, StorageError>
-    {
-        let v1 = ScanJournalTable::scan_journals(partition_store, range.clone())?
-            .map(|x| x.map(|(id, entry)| (id, ScannedEntry::V1(entry))));
+        f: F,
+    ) -> Result<impl Future<Output = restate_storage_api::Result<()>> + Send, StorageError> {
+        // these two iterators can run concurrently in theory.
```

Comment:
<img width="244" height="200" alt="image" src="https://github.com/user-attachments/assets/b2acc5cc-77f9-4fe4-8952-202c6377cd62" />

Reply:
cc @slinkydeveloper any plans to migrate journal rows?